### PR TITLE
Refactor fill_form to be able to do things one page at a time.

### PR DESCRIPTION
**Pull request by Wildebeest**

### CODE REVIEW

File: .ruby-gemset [hunk 1]
```diff
@@ -0,0 +1 @@
+prawn-fillform
```
Comment (alexgerstein) [inline .ruby-gemset:1]: What's this change?

File: Gemfile [hunk 2]
```diff
@@ -2,5 +2,4 @@ source "http://rubygems.org"
 
 # Specify your gem's dependencies in prawn-fillform.gemspec
 #
-gem 'prawn', :require => 'prawn'
```
Comment (alexgerstein) [inline Gemfile:5]: We still need prawn, no?

File: lib/prawn-fillform.rb [hunk 3]
```diff
@@ -263,130 +244,145 @@ def acroform_field_names
...
-            if dictionary[:Parent]
-              deref(deref(dictionary[:Parent])[:Kids]).map { |kid| deref(kid) }.select { |kid| kid[:P] == page.dictionary }
-            else
-              [dictionary]
-            end
-          end.each do |dictionary|
-            next unless deref(dictionary[:Type]) == :Annot and deref(dictionary[:Subtype]) == :Widget
+        acroform[page_number] = acroform_fields_for_page(page)
+      end
+    end
```
Comment (alexgerstein) [inline lib/prawn-fillform.rb:251]: Does this need to return acroform here?

File: lib/prawn-fillform.rb [hunk 4]
```diff
@@ -263,130 +244,145 @@ def acroform_field_names
...
-                bounding_box([x_position, y_position + 2], :width => width, :height => height) do
-                  table([value.split('')], :cell_style => { :borders => [] }, :column_widths => width/field.max_length ) do
-                    rows(0).height = field.height
-                  end
+    # Found by manual adjustment until it looked right.
+    FILLFORM_X_OFFSET = -34
+    FILLFORM_Y_OFFSET = -38
+
+    def fill_form_page_with(data, x_offset: FILLFORM_X_OFFSET, y_offset: FILLFORM_Y_OFFSET)
+      acroform_fields_for_page(page).each do |field|
```
Comment (alexgerstein) [inline lib/prawn-fillform.rb:306]: Where does the page variable come from?

File: lib/prawn-fillform.rb [hunk 5]
```diff
@@ -263,130 +244,145 @@ def acroform_field_names
...
+                image value, image_options
               end
             end
           end
         end
       end
+    end
+
+    def fill_form_with(data={})
+      state.pages.length.times do |number|
```
Comment (alexgerstein) [inline lib/prawn-fillform.rb]: `state.pages.each_with_index`

File: .ruby-version [hunk 6]
```diff
@@ -0,0 +1 @@
+ruby-2.2.3
```
Comment (alexgerstein) [inline .ruby-version:1]: And this one?

File: Gemfile (hunk 2 above)
Comment (Wildebeest) [inline Gemfile:5]: It's in the gemspec, so this is redundant.

File: lib/prawn-fillform.rb (hunk 3 above)
Comment (Wildebeest) [inline lib/prawn-fillform.rb:251]: Yes.

File: lib/prawn-fillform.rb (hunk 4 above)
Comment (Wildebeest) [inline lib/prawn-fillform.rb:306]: It's defined by `Document`, which this gets mixed into.

File: lib/prawn-fillform.rb (hunk 5 above)
Comment (Wildebeest) [inline lib/prawn-fillform.rb]: Sure. I don't actually need the page though.

File: .ruby-gemset (hunk 1 above)
Comment (Wildebeest) [inline .ruby-gemset:1]: This and .ruby-version are here to make this repo play better with rvm. This tells RVM to make a new gemset for this repo.

File: .ruby-version (hunk 6 above)
Comment (Wildebeest) [inline .ruby-version:1]: And this tells RVM which version of ruby to use.

File: .ruby-gemset (hunk 1 above)
Comment (alexgerstein) [inline .ruby-gemset:1]: Cool

File: lib/prawn-fillform.rb (hunk 4 above)
Comment (alexgerstein) [inline lib/prawn-fillform.rb:306]: Oh yeah. I realized that as I kept going. Thanks!

**Comment by alexgerstein** [comment]
@zp-alfred Accept!

**Comment by zp-alfred** [comment]
Accepted by alexgerstein!

**Pull request closed without merging**